File: server/Diskollection/Diskollection/Models/Album.cs
using System.ComponentModel.DataAnnotations;
using System.Diagnostics.CodeAnalysis;

namespace Diskollection.Models
{
    public class Album
    {
        public int Id { get; set; }
        public string Title { get; set; }
        public int? ArtistId { get; set; }
        [AllowNull]
        public Artist? Artist { get; set; }
        [DataType(DataType.Date)]
        public DateTime ReleaseDate { get; set; }
        public string Genre { get; set; }
        [DataType(DataType.ImageUrl)]
        public string ImageURL { get; set; }
        public string Description { get; set; }

        public int Likes { get; set; } = 0;
    }
}

File: client/Diskollection/src/app/shared/models/Album.ts
export interface Album{
    id: string;
    title: string;
    artistId: string;
    artistName: string;
    releaseDate: string;
    genre: string;
    imageURL: string;
    description?: string;
    likes: 0;
}

export interface AlbumDTO{
    id?: string;
    title: string;
    artistId: string;
    releaseDate: string;
    genre: string;
    imageURL?: string;
    description?: string;
}